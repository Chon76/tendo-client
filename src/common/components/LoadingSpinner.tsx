import { Center, Spinner } from 'native-base';

export default function LoadingSpinner() {
  return (
    <Center flex={1}>
      <Spinner></Spinner>
    </Center>
  );
}
