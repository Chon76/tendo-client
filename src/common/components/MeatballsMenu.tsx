import { FontAwesome5 } from '@expo/vector-icons';
import { Icon, Menu, Pressable } from 'native-base';

export function MeatballsMenu({ children }: any) {
  return (
    <Menu
      trigger={(triggerProps) => {
        return (
          <Pressable position="absolute" right={3} zIndex={1} {...triggerProps}>
            <Icon as={FontAwesome5} name="ellipsis-h" size={7} />
          </Pressable>
        );
      }}
    >
      {children}
    </Menu>
  );
}
