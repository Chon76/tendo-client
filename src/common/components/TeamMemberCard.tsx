import { Image, Row, Text } from 'native-base';
import { useSelector } from 'react-redux';
import { User } from '../../types';
export default function TeamMemberCard({
  member,
  showMvp = true,
}: {
  member: User & { role: 'captain' | 'member' };
  showMvp?: boolean;
}) {
  const { userId } = useSelector((state: any) => state.userReducer);

  return (
    <Row
      bg={userId === member.id ? 'primary.500' : 'transparent'}
      justifyContent={'space-between'}
      alignItems={'center'}
      borderRadius="md"
      p={2}
      mb={3}
    >
      <Image
        alt="player's avatar"
        source={member.avatarUrl ? { uri: member.avatarUrl } : require('@common/assets/images/icon.png')}
        size={'xs'}
        borderRadius="md"
      />

      <Text>
        {member.firstName} {member.lastName}
      </Text>

      {showMvp && <Text color={'primary.300'}>MVP: {member.mvps}</Text>}

      <Text>{member.role === 'captain' && '*'}</Text>
    </Row>
  );
}
