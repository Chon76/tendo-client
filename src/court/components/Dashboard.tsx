import { Center, Column, ScrollView, Text } from 'native-base';
import UpcomingFixtures from './UpcomingFixtures';

export default function Dashboard() {
  return (
    <ScrollView bounces={false}>
      <Center m={5}>
        <Column w="full" space={5}>
          <UpcomingFixtures />
          {/* <Center>
            <Text>local rankings...</Text>
          </Center> */}
        </Column>
      </Center>
    </ScrollView>
  );
}
