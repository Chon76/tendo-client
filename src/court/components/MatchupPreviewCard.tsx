import { Column, Image, Pressable, Row, Text } from 'native-base';
import moment from 'moment';

import { Fixture } from '../../types';
import { NavigationProp, useNavigation } from '@react-navigation/native';
import { CourtStackParamList } from '@court/navigation/types';

export default function MatchupPreviewCard({ fixture }: { fixture: Fixture }) {
  const navigation = useNavigation<NavigationProp<CourtStackParamList>>();
  return (
    <Pressable onPress={() => navigation.navigate('Matchup', { id: fixture.id })}>
      <Row
        bg="dark.50"
        _light={{ bg: 'light.50' }}
        justifyContent={'space-between'}
        alignItems={'center'}
        p={4}
        mx={2}
        borderRadius={'md'}
      >
        <Column space={1}>
          <Row space={1}>
            <Image alt="team logo" source={require('@common/assets/images/icon.png')} size="2xs" />
            <Text>{fixture.teamA.name}</Text>
          </Row>
          <Row space={1}>
            <Image alt="team logo" source={require('@common/assets/images/icon.png')} size="2xs" />
            <Text>{fixture.teamB.name}</Text>
          </Row>
        </Column>
        <Column alignItems={'center'}>
          <Text textAlign={'center'}>{moment(fixture.datetime).calendar({ sameElse: 'YYYY-MM-DD [\n] h:mm' })}</Text>
        </Column>
      </Row>
    </Pressable>
  );
}
