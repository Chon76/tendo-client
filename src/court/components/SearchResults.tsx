import { CourtStackParamList } from '@court/navigation/types';
import { NavigationProp, useNavigation } from '@react-navigation/native';
import { Center, Pressable, Text } from 'native-base';

export default function SearchResults({ results }: any) {
  const navigation = useNavigation<NavigationProp<CourtStackParamList>>();

  const navigateToTeam = (team: any) => {
    navigation.navigate('TeamDetails', { id: team.id });
  };
  return (
    <>
      <Center mt={5}>
        <Text>Teams</Text>
      </Center>
      {results &&
        results.teams.map((team: any) => (
          <Pressable key={team.id} onPress={() => navigateToTeam(team)}>
            <Text>
              {team.location} {team.name}
            </Text>
          </Pressable>
        ))}
    </>
  );
}
