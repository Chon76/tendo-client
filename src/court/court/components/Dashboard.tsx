import { Center, Column, ScrollView } from 'native-base';
import UpcomingFixtures from './UpcomingFixtures';

export default function Dashboard() {
  return (
    <ScrollView>
      <Center mt={2} mx={5}>
        <Column w="full">
          <UpcomingFixtures />
        </Column>
      </Center>
    </ScrollView>
  );
}
