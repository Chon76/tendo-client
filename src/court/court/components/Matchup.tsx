import { Column, Image, Pressable, Row, Text } from 'native-base';
import moment from 'moment';

import { Fixture } from '../../../types';

export default function Matchup({ fixture }: { fixture: Fixture }) {
  return (
    <Pressable onPress={() => console.log(fixture.id)}>
      <Row
        bg="dark.50"
        _light={{ bg: 'light.50' }}
        justifyContent={'space-between'}
        alignItems={'center'}
        p={4}
        mx={2}
        borderRadius={'md'}
      >
        <Column space={1}>
          <Row space={1}>
            <Image alt="team logo" source={require('@common/assets/images/icon.png')} size="2xs" />
            <Text>{fixture.teamA.name}</Text>
          </Row>
          <Row space={1}>
            <Image alt="team logo" source={require('@common/assets/images/icon.png')} size="2xs" />
            <Text>{fixture.teamB.name}</Text>
          </Row>
        </Column>
        <Column alignItems={'center'}>
          <Text textAlign={'center'}>{moment(fixture.datetime).calendar({ sameElse: 'YYYY-MM-DD [\n] h:mm' })}</Text>
        </Column>
      </Row>
    </Pressable>
  );
}
