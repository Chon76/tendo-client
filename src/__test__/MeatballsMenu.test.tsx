import { MeatballsMenu } from '@common/components/MeatballsMenu';
import { render } from './test-utils';

describe('header', () => {
  it('renders correctly', () => {
    const meatballsMEnu = render(<MeatballsMenu />);

    expect(meatballsMEnu.toJSON()).toMatchSnapshot();
  });
});
