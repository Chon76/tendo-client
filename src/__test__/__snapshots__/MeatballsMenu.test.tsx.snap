// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    aria-haspopup="menu"
    collapsable={false}
    dataSet={Object {}}
    focusable={true}
    nativeID="react-aria-1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onKeyDownCapture={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": 12,
        "zIndex": 1,
      }
    }
  >
    <Text />
  </View>
</RNCSafeAreaProvider>
`;
